### PR TITLE
Add Image tests

### PR DESCRIPTION
**Pull request by Watson1978**
This patch will improve coverage of rmimage.c

* Before
  - Line coverage : 84.4 %
  - Functions : 98.4 %

* After
  - Line coverage : 93.0 %
  - Functions : 99.2 %

Related to https://github.com/rmagick/rmagick/issues/542

### CODE REVIEW

**Comment by Watson1978** [comment]
@dlemstra Do you know failure reason ? it will be caused on IM 6.7.

```
===============================================================================
Failure:
test_iptc_profile(Image_Attributes_UT)
/home/travis/build/rmagick/rmagick/test/Image_attributes.rb:405:in `test_iptc_profile'
     402:     assert_nothing_raised { @img.iptc_profile }
     403:     assert_nil(@img.iptc_profile)
     404:     assert_nothing_raised { @img.iptc_profile = 'xxx' }
  => 405:     assert_equal('xxx', @img.iptc_profile)
     406:     assert_raise(TypeError) { @img.iptc_profile = 2 }
     407:   end
     408: 
<"xxx"> expected but was
<nil>
```

**Comment by Watson1978** [comment]
SEGV with IM 6.9 😫

**Comment by Watson1978** [comment]
@dlemstra Do you know anything about the IM 6.9 API used in Image#add_profile method?

**Comment by dlemstra** [comment]
Will take a look tonight

**Comment by dlemstra** [comment]
The SEGV appears to be an upstream issue: https://github.com/ImageMagick/ImageMagick6/issues/41

**Comment by dlemstra** [comment]
And I suspect (not tested) that the 6.7.7 is a casing issue. The profile is added in lowercase but retrieved in uppercase. Recent versions of IM will change the value to lowercase and that is why the tests pass there. This is resolved in PR: #556. And you might also want to add a test for `@img.icc_profile` here.

File: test/Image1.rb [hunk 1]
```diff
@@ -54,7 +67,7 @@ def test_adaptive_blur_channel
     assert_nothing_raised { @img.adaptive_blur_channel(3, 2) }
     assert_nothing_raised { @img.adaptive_blur_channel(3, 2, Magick::RedChannel) }
     assert_nothing_raised { @img.adaptive_blur_channel(3, 2, Magick::RedChannel, Magick::BlueChannel) }
-    assert_raise(ArgumentError) { @img.adaptive_blur(3, 2, 2) }
+    assert_raise(TypeError) { @img.adaptive_blur_channel(3, 2, 2) }
```
Comment (dlemstra) [inline test/Image1.rb]: Good catch :)

File: test/Image1.rb [hunk 2]
```diff
@@ -285,6 +332,16 @@ def test_border
     assert_raise(FreezeError) { @img.border!(2, 2, 'red') }
   end
 
+  def test_capture
+    # assert_nothing_raised { Magick::Image.capture }
```
Comment (dlemstra) [inline test/Image1.rb]: These need some attention in another PR?

File: test/Image3.rb [hunk 3]
```diff
@@ -505,8 +483,15 @@ def test_selective_blur_channel
     assert_nothing_raised { @img.selective_blur_channel(0, 1, '10%', Magick::RedChannel) }
     assert_nothing_raised { @img.selective_blur_channel(0, 1, '10%', Magick::RedChannel, Magick::BlueChannel) }
     assert_nothing_raised { @img.selective_blur_channel(0, 1, '10%', Magick::RedChannel, Magick::BlueChannel, Magick::GreenChannel) }
-    # not enough arguments
```
Comment (dlemstra) [inline test/Image3.rb]: 👍

File: test/Image_attributes.rb [hunk 4]
```diff
@@ -204,63 +212,9 @@ def test_compose
...
-    assert_nothing_raised { @img.compose = Magick::SrcOverCompositeOp }
-    assert_nothing_raised { @img.compose = Magick::SubtractCompositeOp }
-    assert_nothing_raised { @img.compose = Magick::ThresholdCompositeOp }
-    assert_nothing_raised { @img.compose = Magick::XorCompositeOp }
-    assert_nothing_raised { @img.compose = Magick::MathematicsCompositeOp }
-    assert_nothing_raised { @img.compose = Magick::DivideSrcCompositeOp }
-    assert_nothing_raised { @img.compose = Magick::MinusSrcCompositeOp }
-    assert_nothing_raised { @img.compose = Magick::DarkenIntensityCompositeOp }
-    assert_nothing_raised { @img.compose = Magick::LightenIntensityCompositeOp }
+    Magick::CompositeOperator.values do |composite|
```
Comment (dlemstra) [inline test/Image_attributes.rb]: Nice refactoring 🥇

File: test/Image1.rb (hunk 2 above)
Comment (Watson1978) [reply test/Image1.rb]: Yep, seems Magick::Image.capture does not work... It needs to fix another PR.

Comment (dlemstra) [reply test/Image1.rb]: Wonder how you can fix this. The method calls `display=XOpenDisplay(image_info->server_name);` and I don't think that will work in a CI build.

Comment (Watson1978) [reply test/Image1.rb]: Filed issue in https://github.com/rmagick/rmagick/issues/560

Comment (Watson1978) [reply test/Image1.rb]: > Wonder how you can fix this. The method calls display=XOpenDisplay(image_info->server_name); and I don't think that will work in a CI build.

Ah, i see

**Comment by Watson1978** [comment]
@dlemstra Seems the test still failed with IM 6.9
Do you know the reason ?

```
===============================================================================
Failure: <Magick::ImageMagickError> exception expected but none was thrown.
test_add_delete_profile(Image1_UT)
/home/circleci/repo/test/Image1.rb:157:in `test_add_delete_profile'
     154:   def test_add_delete_profile
     155:     img = Magick::Image.read(IMAGES_DIR + '/Button_0.gif').first
     156:     assert_nothing_raised { img.add_profile(File.join(__dir__, 'cmyk.icm')) }
  => 157:     assert_raise(Magick::ImageMagickError) { img.add_profile(File.join(__dir__, 'srgb.icm')) }
     158: 
     159:     img.each_profile { |name, _value| assert_equal('icc', name) }
     160:     assert_nothing_raised { img.delete_profile('icc') }
===============================================================================
```

**Comment by dlemstra** [comment]
Looks like the code should be changed to: `assert_nothing_raised { img.add_profile(File.join(__dir__, 'srgb.icm')) }` for recent version of IM. Not sure when this was changed though.

**Comment by Watson1978** [comment]
Sounds it remove the tests that depend on IM behavior.

**Pull request merged**